src/components/TestingModelGridView.tsx

```diff
@@ -70,8 +74,32 @@ const TestingModelGridView = () => {
   const [{ languageId }] = useSettings();
   const makeCodeLang = getMakeCodeLang(languageId);
 
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  const continueToEditor = useCallback(async () => {
+    await openEditor();
+    onClose();
+  }, [onClose, openEditor]);
+
+  const maybeOpenEditor = useCallback(() => {
+    // Open editor if device is not a V1, otherwise show warning dialog.
+    const bluetoothConnection = getBluetoothConnection();
+    if (bluetoothConnection) {
+      if (bluetoothConnection.getBoardVersion() === "V1") {
+        return onOpen();
+      }
+    }
```

Comment:
Might be nice to hide the bluetooth-specific nature of this behind something in the hook as we might do V1 radio bridge at some point.

Reply:
Done here. A bit more work to hold on to the radio remote board version [5dc85ea](https://github.com/microbit-foundation/ml-trainer/pull/358/commits/5dc85ea44d2b03748b9b729314d8a38b64e233c4)

`radioRemoteBoardVersion` is currently not cleared if you go back to the home page and start a new session. It's only cleared on a page refresh.

Reply:
We're going to ponder this on a separate ticket.